### PR TITLE
Created ShareRepo and Tests for ShareRepo

### PR DESCRIPTION
**Pull request by myothiha09**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @myothiha09, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: box-share-sdk/src/main/java/com/box/androidsdk/share/sharerepo/ShareRepo.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private final MutableLiveData<BoxResponse<BoxCollaborationItem>> mShareItem = new MutableLiveData<BoxResponse<BoxCollaborationItem>>();
+
+    private final MutableLiveData<BoxResponse<BoxResponseBatch>> mInvitingCollabsChecker = new MutableLiveData<BoxResponse<BoxResponseBatch>>();
+
+    public ShareRepo(ShareController controller) {
+        this.mController = controller;
+    }
+
+    @Override
+    public LiveData<BoxResponse<BoxIteratorInvitees>> getInvitees(BoxCollaborationItem boxCollaborationItem, String filter) {
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/sharerepo/ShareRepo.java]: Add javadoc to all public methods.

Comment (myothiha09) [reply box-share-sdk/src/main/java/com/box/androidsdk/share/sharerepo/ShareRepo.java]: Got it.

File: box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        mShareRepo.addCollabsApi(boxCollaborationItem, selectedRole, emails);
+    }
+
+    public void getInviteesApi(BoxCollaborationItem boxCollaborationItem, String filter) {
+        mShareRepo.getInviteesApi(boxCollaborationItem, filter);
+    }
+
+
+
+    public LiveData<DataWrapper<BoxCollaborationItem>> getmFetchRoleItem() {
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java]: Update all AS generated methods to not include 'm' char

File: box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    private DataWrapper<BoxIteratorInvitees> createGetInviteesItemData(BoxResponse<BoxIteratorInvitees> response) {
+        final DataWrapper<BoxIteratorInvitees> data = new DataWrapper<BoxIteratorInvitees>();
+        if (response.isSuccess()) {
+            final BoxIteratorInvitees invitees = response.getResult();
+            data.success(invitees);
+        } else {
+            BoxException boxException = (BoxException) response.getException();
+            int responseCode = boxException.getResponseCode();
+            int strCode = -1;
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java]: rename to errorStr or errorMsg

Comment (nyjalusc) [reply box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java]: add a default error string.

File: box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+            data.failure(strCode);
+        }
+        return data;
+    }
+
+    private InviteCollaboratorsDataWrapper createAddCollabItemData(BoxResponse<BoxResponseBatch> response) {
+        return handleCollaboratorsInvited(response.getResult());
+    }
+
+    private InviteCollaboratorsDataWrapper handleCollaboratorsInvited(BoxResponseBatch responses) {
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java]: Refactor this method into smaller methods and name it correctly.

File: box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsDataWrapper.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package com.box.androidsdk.share.vm;
+
+/**
+ * A special class to hold data for inviting a new Collaborator
+ */
+public class InviteCollaboratorsDataWrapper {
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsDataWrapper.java]: Inherit from DataWrapper if possible.

File: box-share-sdk/src/main/java/com/box/androidsdk/share/vm/BaseVM.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    protected final BaseShareRepo mShareRepo;
+    protected final MutableLiveData<BoxCollaborationItem> mShareItem;
+
+    public BaseVM(BaseShareRepo shareRepo, BoxCollaborationItem shareItem) {
+        this.mShareRepo = shareRepo;
+        this.mShareItem = new MutableLiveData<BoxCollaborationItem>();
+        this.mShareItem.postValue(shareItem);
+    }
+
+    public MutableLiveData<BoxCollaborationItem> getmShareItem() {
```
Comment (nyjalusc) [inline box-share-sdk/src/main/java/com/box/androidsdk/share/vm/BaseVM.java]: return should always be `LiveData`

**Comment by nyjalusc** [comment]
Create a separate feature branch (based on master) and create PR's for that.
Break down this PR into smaller PR's ideally into repo, vm's, ui and moving code to legacy packages.

**Pull request closed without merging**